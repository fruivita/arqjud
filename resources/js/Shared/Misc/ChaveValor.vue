<!--
    Exibe o par chave e valor estilizado como um 'input' desabilitado.

    Útil para exibir valores imutáveis, presentes na página para informação.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import MensagemErro from '@/Shared/Forms/MensagemErro.vue';
import Icone from '@/Shared/Icones/Icone.vue';

defineProps({
    icone: { type: String },
    chave: { type: String, required: true },
    valor: { type: [Number, String] },
    erro: { type: String },
});
</script>

<template>
    <div class="space-y-1">
        <span class="font-bold">{{ chave }}</span>

        <p
            class="flex space-x-4 rounded-lg border border-dashed border-primaria-500 bg-primaria-50 p-2 transition duration-300 dark:bg-secundaria-900 dark:text-primaria-50"
        >
            <Icone v-if="icone" :nome="icone" />

            <span v-if="valor">{{ valor }}</span>
        </p>

        <MensagemErro v-if="erro" :erro="erro" />
    </div>
</template>
