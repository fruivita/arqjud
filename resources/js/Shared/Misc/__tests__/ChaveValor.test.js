/**
 * Testes para o componente ChaveValor.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import ChaveValor from '@/Shared/Misc/ChaveValor.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(ChaveValor, { ...options });
    };
});

describe('ChaveValor', () => {
    // Caminho feliz
    test('propriedades do componente estão definidas', () => {
        expect(ChaveValor.props).toMatchObject({
            icone: { type: String },
            chave: { type: String, required: true },
            valor: { type: [Number, String] },
        });
    });

    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { chave: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente com o valor informado respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { chave: 'foo', valor: 'bar' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente com o ícone respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { icone: 'key', chave: 'foo', valor: 'bar' },
            }).html()
        ).toMatchSnapshot();
    });
});
