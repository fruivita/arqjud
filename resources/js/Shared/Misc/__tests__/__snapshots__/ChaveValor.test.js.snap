// Vitest Snapshot v1

exports[`ChaveValor > renderiza o componente com o valor informado respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><span class=\\"font-bold\\">foo</span>
  <p class=\\"flex space-x-4 rounded-lg border border-dashed border-primaria-500 bg-primaria-50 p-2 transition duration-300 dark:bg-secundaria-900 dark:text-primaria-50\\">
    <!--v-if--><span>bar</span>
  </p>
</div>"
`;

exports[`ChaveValor > renderiza o componente com o Ã­cone respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><span class=\\"font-bold\\">foo</span>
  <p class=\\"flex space-x-4 rounded-lg border border-dashed border-primaria-500 bg-primaria-50 p-2 transition duration-300 dark:bg-secundaria-900 dark:text-primaria-50\\"><svg class=\\"w-6 h-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M0 8a4 4 0 0 1 7.465-2H14a.5.5 0 0 1 .354.146l1.5 1.5a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0L13 9.207l-.646.647a.5.5 0 0 1-.708 0L11 9.207l-.646.647a.5.5 0 0 1-.708 0L9 9.207l-.646.647A.5.5 0 0 1 8 10h-.535A4 4 0 0 1 0 8zm4-3a3 3 0 1 0 2.712 4.285A.5.5 0 0 1 7.163 9h.63l.853-.854a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.793-.793-1-1h-6.63a.5.5 0 0 1-.451-.285A3 3 0 0 0 4 5z\\"></path>
      <path d=\\"M4 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0z\\"></path>
    </svg><span>bar</span></p>
</div>"
`;

exports[`ChaveValor > renderiza o componente respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><span class=\\"font-bold\\">foo</span>
  <p class=\\"flex space-x-4 rounded-lg border border-dashed border-primaria-500 bg-primaria-50 p-2 transition duration-300 dark:bg-secundaria-900 dark:text-primaria-50\\">
    <!--v-if-->
    <!--v-if-->
  </p>
</div>"
`;
